app/src/main/java/org/simple/clinic/AppDatabase.kt

```diff
@@ -77,4 +78,43 @@ abstract class AppDatabase : RoomDatabase() {
         """)
     }
   }
+
+  class Migration_4_5 : Migration(4, 5) {
+
+    override fun migrate(database: SupportSQLiteDatabase) {
+      // Update local searchable name in the Patient table to strip out the newly added characters
+      database.inTransaction {
+        compileStatement("""update "Patient" set "searchableName"=? where "uuid"=?""")
+            .use { statement ->
+              query("""select "uuid","fullName" from "Patient"""")
+                  .use { cursor ->
+                    val uuidIndex = cursor.getColumnIndex("uuid")
+                    val fullNameIndex = cursor.getColumnIndex("fullName")
+
+                    generateSequence { if (cursor.moveToNext()) cursor else null }
+                        .map { it.getString(uuidIndex) to it.getString(fullNameIndex) }
+                        .map { (uuid, fullName) -> uuid to convertNameToSearchableForm(fullName) }
+                        .forEach { (uuid, searchableName) ->
+                          statement.bindString(1, searchableName)
+                          statement.bindString(2, uuid)
+                          statement.executeUpdateDelete()
+                        }
+                  }
+            }
```

Comment:
These `use` blocks are adding a lot of nesting + I don't think you're taking advantage of them. Will discuss with you in person.

Reply:
The `use` blocks automatically close any `Closeable` instances, like the `Cursor` and the `SQLiteStatement` for you.